extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -716,6 +787,23 @@ export class AmpAutocomplete extends AMP.BaseElement {
     });
   }
 
+  /**
+   * Calculates the total offset distance from the given element to the
+   * container_. This assumes the given parameter is a descendent of the
+   * results container.
+   *
+   * @param {!Element} element
+   * @return {number}
+   * @private
+   */
+  distanceToContainerTop_(element) {
+    const {offsetTop: top, parentElement: parent} = element;
+    if (parent === this.container_) {
+      return top;
+    }
+    return top + this.distanceToContainerTop_(parent);
```

Comment:
Is this the same as `this.container_.getBoundingClientRect() - element.getBoundingClientRect().top`? You might not need to traverse the node tree

Reply:
Doing that leads to twitchy behavior... I think there is a simpler way like you are suggesting, but I get confused by the way the DOM measures elements that are contained inside each other. Maybe we can take a look offline.

Reply:
Hmm strange, I wonder if it's a rounding error since those measurements are relative to the viewport scroll. Sounds good

Reply:
A couple notes after playing around with this more this afternoon:
- The helper function actually doesn't work for multiply nested divs because of the weird logic from intermediary elements. i.e. it scrolls properly for `category > item` but not `category > category > item`
- However I replaced the helper function with this math: `element.closest(container > category).offsetTop` + `element.offsetTop`. The two elements aren't necessarily directly related but this works **iff**...
- The `position: relative` style is missing from any intermediary parents so they default to `position: static`. This then disqualifies those elements from factoring into the [`element.offsetTop`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetTop) value calculation. 
- As a consequence I went ahead and changed the CSS selector from
```
.i-amphtml-autocomplete-item, .i-amphtml-autocomplete-category {
  position: relative;
}
```
to `.i-amphtml-autocomplete-results > *` but I'm not sure that is ideal. Also open to other ways of specifying the subset I described or trying to figure out another solution. I think this solution is likely more susceptible to accidental breakage.